store-failures

```diff
@@ -80,6 +80,7 @@ def init_db(cursor):
                     (testname TEXT NOT NULL,
                      retry_reason TEXT,
                      run INTEGER NOT NULL,
+                     UNIQUE (testname, retry_reason, run)
```

Comment:
Thanks @martinpitt for the fixes and improvements!
I am not convinced it we want `Avoid duplicates in Failures ` commit - we are interested to see, how many times test fails for statistical reasons. So having the same record twice makes sense, as it points to the fact, it was retried twice and not only once. We could put there retry number or some other identifier, but I am not sure if it is really needed.

Reply:
That's a good point, yes -- so please kick it out again on a rebase. It just somehow needs to make sure that scanning a log twice doesn't end up duplicating these entries. So maybe the transaction should start with a 

    DELETE FROM Failures WHERE run = run_id

?